lib/utils/remote-build.ts

```diff
@@ -101,31 +111,63 @@ export async function startRemoteBuild(build: RemoteBuild): Promise<void> {
 		rl.on('SIGINT', () => process.emit('SIGINT' as any));
 	}
 
-	return new Bluebird((resolve, reject) => {
-		// Setup interrupt handlers so we can cancel the build if the user presses
-		// ctrl+c
-
-		// This is necessary because the `exit-hook` module is used by several
-		// dependencies, and will exit without calling the following handler.
-		// Once https://github.com/balena-io/balena-cli/issues/867 has been solved,
-		// we are free to (and definitely should) remove the below line
-		process.removeAllListeners('SIGINT');
-		process.on('SIGINT', () => {
-			process.stderr.write('Received SIGINT, cleaning up. Please wait.\n');
-			cancelBuildIfNecessary(build).then(() => {
-				stream.end();
-				process.exit(130);
+	if (!build.opts.headless) {
+		return new Bluebird((resolve, reject) => {
+			// Setup interrupt handlers so we can cancel the build if the user presses
+			// ctrl+c
+
+			// This is necessary because the `exit-hook` module is used by several
+			// dependencies, and will exit without calling the following handler.
+			// Once https://github.com/balena-io/balena-cli/issues/867 has been solved,
+			// we are free to (and definitely should) remove the below line
+			process.removeAllListeners('SIGINT');
+			process.on('SIGINT', () => {
+				process.stderr.write('Received SIGINT, cleaning up. Please wait.\n');
+				cancelBuildIfNecessary(build).then(() => {
+					stream.end();
+					process.exit(130);
+				});
 			});
+
+			stream.on('data', getBuilderMessageHandler(build));
+			stream.on('end', resolve);
+			stream.on('error', reject);
+		}).then(() => {
+			if (build.hadError) {
+				throw new RemoteBuildFailedError();
+			}
 		});
+	}
 
-		stream.on('data', getBuilderMessageHandler(build));
-		stream.on('end', resolve);
-		stream.on('error', reject);
-	}).then(() => {
-		if (build.hadError) {
-			throw new RemoteBuildFailedError();
-		}
-	});
+	// We're running a non-headless build, which means we'll
+	// get a single object back, detailing if the build has
+	// been started
+	let result: HeadlessBuilderMessage;
+	try {
+		const response = await streamToPromise(stream);
+		result = JSON.parse(response.toString());
+	} catch (e) {
+		throw new Error(
+			'There was an error reading the response from the remote builder',
+		);
+	}
+	handleHeadlessBuildMessage(result);
+}
+
+function handleHeadlessBuildMessage(message: HeadlessBuilderMessage) {
```

Comment:
Not sure it's worth having this as a separate function, given it's not called from anywhere else and doesn't really make the calling function that much bigger.